CMakeLists.txt

```diff
@@ -68,8 +69,9 @@ set(exported_libraries
 )
 catkin_package(
   INCLUDE_DIRS include
-  LIBRARIES ${exported_libraries} ${YAML_CPP_LIBRARIES}
+  LIBRARIES ${exported_libraries}
   CATKIN_DEPENDS ${CATKIN_PKGS}
+  DEPENDS yaml-cpp
```

Comment:
You should depend on the yaml_cpp_catkin package we have. It adds additional functionalities like the option load of a parameter, the redefinition of the exception error, etc.

If you do it like this you should also update the package.xml to reflect this "depends"

Reply:
It is actually already there ;)

Reply:
So if you depend on yaml_cpp_catkin you should not add yaml-cpp to the package.xml or as dependency in the catkin package because it is already (or should already be there in the yaml_cpp_catkin)